File: GO/binSearch.go
// You can edit this code!
// Click here and start typing.
package main

import (
	"fmt"
	"sort"
)

func bin_search(arr []int, tar int) int {
	left := 0
	right := len(arr) - 1
	i := (left + right) / 2

	for left <= i && right >= i {
		fmt.Printf("Проверка элемента в массиве по индексу %d\n", i)

		if arr[i] == tar {
			fmt.Printf("%d == %d\n", arr[i], tar)
			return i
		} else if arr[i] < tar {
			fmt.Printf("%d < %d\n", arr[i], tar)
			i++
		} else {
			fmt.Printf("%d > %d\n", arr[i], tar)
			i--
		}
	}
	return -1
}

func main() {
	arr := []int{45, 34, 78, 52, 12, 88, 199, 1, 3}
	sort.Ints(arr)
	fmt.Println("Отсортированный массив:", arr)

	tar := 199
	result := bin_search(arr, tar)

	if result != -1 {
		fmt.Printf("Элемент найден: индекс в массиве %d, элемент %d\n", result, arr[result])
	} else {
		fmt.Println("Элемент не найден")
	}
}

File: PYTHON/binsearch.py
def bin_search(arr, tar):
    left = 0
    right = len(arr) - 1
    i = int((left + right) / 2)

    while bool((left <= i) & (right >= i)):
        print(f"Проверка элемента в массиве по индексу {i}")


        if arr[i] == tar:
            print(f"{arr[i]} == {tar}")
            return i
        elif arr[i] < tar:
            print(f"{arr[i]} < {tar}")
            i+=1
        else:
            print(f"{arr[i]} > {tar}")
            i-=1
    return -1


if __name__ == "__main__":
    arr=[45, 34, 78, 52, 12, 88, 199, 1, 3]
    arr.sort()
    print("Отсортированный массив:", arr)

    tar=199
    result = bin_search(arr, tar)

    print(f"Элемент найден: индекс в массиве {result}, элемент {arr[result]}")
